app/models/zone.rb

```diff
@@ -170,37 +159,10 @@ def availability_zones
     ems_clouds.flat_map(&:availability_zones)
   end
 
-  def vms_and_templates
-    MiqPreloader.preload(self, :ext_management_systems => :vms_and_templates)
-    ext_management_systems.flat_map(&:vms_and_templates)
-  end
-
-  def vms
-    MiqPreloader.preload(self, :ext_management_systems => :vms)
-    ext_management_systems.flat_map(&:vms)
-  end
-
   def self.vms_without_a_zone
     Vm.where(:ems_id => nil).to_a
   end
 
-  def miq_templates
-    MiqPreloader.preload(self, :ext_management_systems => :miq_templates)
-    ext_management_systems.flat_map(&:miq_templates)
-  end
-
-  def vm_or_template_ids
```

Comment:
Think we played with inflections. so the singular of `vms_and_templates` is `vm_or_template`... a little too cute for my taste but long story short `vm_or_template_ids` does exist "out of box" (for a very warped box)